modules/rank-eval/build.gradle

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+esplugin {
+  description 'The Rank Eval module adds APIs to evaluate ranking quality.'
+  classname 'org.elasticsearch.index.rankeval.RankEvalPlugin'
+}
+
+dependencies {
+  compile "com.github.spullara.mustache.java:compiler:0.9.1"
+  compile project(':modules:lang-mustache')
```

Comment:
> We can't have modules depending on other modules. 

That pretty much confirms what I already thought.

> Why do you need a dependency directly on mustache scripting? Should it just be on 
> script templates?

Why I think I need mustache templating for this needs a longer explanation of the thinking behind the approach taken in the PR:

Imagine a user having implemented a search box. The way I've seen this done in the past usually followed a pattern similar to the following: Some UI with a search box to get the user to enter their query. Behind that is a bit of application logic that turns the actual query string the user submitted into something the search backend understands. Often this piece of application logic also adds more information to the final query: This could be time of day to emphasize recent results, it could be operating system to return pricier results for Apple users, it could be estimated geo location to emphasize results closer to the users's location.

The problem I've seen people run into is how to combine all those factors into a ranking function that provides relevant results.

This PR tries to address this problem by providing a means to compare multiple different options of combining these ranking factors by means of a template query based on their performance on a pre-defined set of queries with labeled results.

Hope this helps a bit, working on a draft for the actual REST endpoint for better clarity.